pkg/image/admission/admission.go

```diff
@@ -0,0 +1,196 @@
+package admission
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+
+	kadmission "k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+	limitranger "k8s.io/kubernetes/plugin/pkg/admission/limitranger"
+
+	"github.com/openshift/origin/pkg/client"
+	oadmission "github.com/openshift/origin/pkg/cmd/server/admission"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	PluginName = "ImageLimitRange"
+)
+
+func init() {
+	kadmission.RegisterPlugin(PluginName, func(client clientset.Interface, config io.Reader) (kadmission.Interface, error) {
+		plugin, err := NewImageLimitRangerPlugin(client, config)
+		if err != nil {
+			return nil, err
+		}
+		return plugin, nil
+	})
+}
+
+// imageLimitRangerPlugin is the admission plugin.
+type imageLimitRangerPlugin struct {
+	*kadmission.Handler
+	osClient    client.Interface
+	limitRanger kadmission.Interface
+
+	imageStreamReflector *cache.Reflector
+	imageStreamStore     cache.Store
+	stopChan             chan struct{}
+}
+
+// imageLimitRangerPlugin implements the LimitRangerActions interface.
+var _ limitranger.LimitRangerActions = &imageLimitRangerPlugin{}
+
+var _ = oadmission.WantsOpenshiftClient(&imageLimitRangerPlugin{})
+var _ = oadmission.Validator(&imageLimitRangerPlugin{})
+
+// NewImageLimitRangerPlugin provides a new imageLimitRangerPlugin.  Also see SetOpenshiftClient.
+func NewImageLimitRangerPlugin(client clientset.Interface, config io.Reader) (*imageLimitRangerPlugin, error) {
+	plugin := &imageLimitRangerPlugin{
+		Handler: kadmission.NewHandler(kadmission.Create, kadmission.Update),
+		// TODO: share the cache with resource quota controller
+		imageStreamStore: cache.NewStore(cache.MetaNamespaceKeyFunc),
```

Comment:
It's passed to a reflector. Not needed in `imageLimitRangePlugin`. I'll remove it.